enterprise/models.py

```diff
@@ -1360,15 +1393,28 @@ def __repr__(self):
 
     def save(self, *args, **kwargs):  # pylint: disable=arguments-differ
         """
-        Override of model save method to dynamically generate the password field and perform additional validation.
+        Override of model save method to handle encryption logic for password fields.
         """
+        # These will only get passed in from the Django Admin Form for this model.
+        decrypted_password = getattr(self, 'decrypted_password', None)
+        decrypted_sftp_password = getattr(self, 'decrypted_sftp_password', None)
         if not self.pk:
             self.initialization_vector = utils.generate_aes_initialization_vector()
+            decrypted_password = decrypted_password or get_random_string(length=32)
+
+        if decrypted_password:
             self.password = utils.encrypt_string(
-                get_random_string(length=32),
+                decrypted_password,
                 self.initialization_vector
             )
-            self.full_clean()
+
+        if decrypted_sftp_password:
+            self.sftp_password = utils.encrypt_string(
+                decrypted_sftp_password,
+                self.initialization_vector
+            )
+
+        self.full_clean()
```

Comment:
nit: We will now call self.full_clean() on each save even if we don't manually update the password or sftp password. Probably not a problem but slight overhead to recheck validation. Could wrap in a `if decrypted_password or decrypted_sftp_password`